src/bootstrap/src/core/sanity.rs

```diff
@@ -177,6 +177,41 @@ than building it.
             continue;
         }
 
+        // Check if there exists a built-in target in the list of supported targets.
+        let cur_dir = env::current_dir().unwrap().display().to_string();
+        
+        let supported_target_file = format!("{cur_dir}/compiler/rustc_target/src/spec/mod.rs");
+        let contents = fs::read_to_string(supported_target_file)
+            .expect("Should have been able to read the file.");
+
+        let (mut macro_beg, mut macro_end) = (false, false);
+        let mut has_target = false;
+
+        let target_str = target.to_string();
+        for line in contents.lines() {
+            if macro_end { 
+                break; 
+            }
+            if !macro_beg && line.contains("supported_targets!") {
+                macro_beg = true;
+            } else if macro_beg {
```

Comment:
```suggestion
            if !macro_beg {
                macro_beg = line.contains("supported_targets!");
            } else {
```

I don't know if that counts as premature opt;

you could also put a continue in the then block and then reindent the else block?

Reply:
Yeah, indeed this is much cleaner.